File: ClientApp/src/services/product.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Product } from '../models/product.model';
import { environment } from '../environments/environment';


@Injectable({
  providedIn: 'root'
})
export class ProductService {

  private readonly url = "Product";

  constructor(private http: HttpClient) { }

  getAllProducts(): Observable<Product[]>{
    return this.http.get<Product[]>(`${environment.apiUrl}/${this.url}`)
  }

  getProductById(id: string): Observable<Product>{
    return this.http.get<Product>(`${environment.apiUrl}/${this.url}/${id}`);
  }

  addProduct(product: Product): Observable<void>{
    return this.http.post<void>(`${environment.apiUrl}/${this.url}`, product);
  }

  editProduct(id: string, product: Product): Observable<Product>{
    return this.http.put<Product>(`${environment.apiUrl}/${this.url}/${id}`, product);
  }

  deleteProduct(id: string): Observable<void>{
    return this.http.delete<void>(`${environment.apiUrl}/${this.url}/${id}`);
  }
}

File: Services/ProductService.cs
using GamersChat.Models;
using GamersChat.Repositories;
using GamersChat.Repositories.Interfaces;

namespace GamersChat.Services
{
    public class ProductService
    {
        private readonly IProductRepository _productRepository;

        public ProductService(IProductRepository productRepository)
        {
            _productRepository = productRepository;
        }

        public void AddProduct(Product product)
        {
            _productRepository.AddProduct(product);
        }

        public Product EditProduct(Product product)
        {
            return this._productRepository.EditProduct(product);
        }

        public void DeleteProduct(Guid productId)
        {
            _productRepository.DeleteProduct(productId);
        }

        public Product GetProduct(Guid productId)
        {
            return _productRepository.GetProduct(productId);
        }

        public IEnumerable<Product> GetAllProducts()
        {
            return _productRepository.GetAllProducts();
        }
    }
}
